src/dmd/dsymbol.d

```diff
@@ -2270,3 +2268,96 @@ extern (C++) final class DsymbolTable : RootObject
         return tab.length;
     }
 }
+
+/**********************************************
+ * ImportC tag symbols sit in a parallel symbol table,
+ * so that this C code works:
+ * ---
+ * struct S { a; };
+ * int S;
+ * struct S s;
+ * ---
+ * But there are relatively few such tag symbols, so that would be
+ * a waste of memory and complexity. An additional problem is we'd like the D side
+ * to find the tag symbols with ordinary lookup, not lookup in both
+ * tables, if the tag symbol is not conflicting with an ordinary symbol.
+ * The solution is to put the tag symbols that conflict into an associative
+ * array, indexed by the address of the ordinary symbol that conflicts with it.
+ * C has no modules, so this associative array is tagSymTab[] in ModuleDeclaration.
+ * A side effect of our approach is that D code cannot access a tag symbol that is
+ * hidden by an ordinary symbol. This is more of a theoretical problem, as nobody
+ * has mentioned it when importing C headers. If someone wants to do it,
+ * too bad so sad. Change the C code.
```

Comment:
Not possible in some cases, i.e: [sigaction](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sigaction.html) is used by druntime in a few places.

Reply:
The ImportC code will work, but the `struct sigaction` won't be accessible from the D code, just the `sigaction` function. Note that druntime renames the struct to `sigaction_t`.

Adding this to the C code:

    typedef struct sigaction sigaction_t;

will resolve the issue, though the user will have to do it.

Reply:
Another solution would be to declare the two sigaction's in separate .h files and compile them separately.

Reply:
I mention that as an example because a good milestone would be for importC to be able to replace the hand-maintained druntime bindings.  For instance, I've ported druntime to AIX and HPUX, but I can't publish the bindings because I don't know if that would break some copyright law.

Reply:
I understand, but the hand portion can be a .h file with the single typedef mentioned above.